File: binding/src/charts/libqhpiemodelmapper.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qhpiemodelmapper.html
pub const qhpiemodelmapper = struct {

    /// New constructs a new QHPieModelMapper object.
    pub fn New() QtC.QHPieModelMapper {
        return qtc.QHPieModelMapper_new();
    }


    /// New2 constructs a new QHPieModelMapper object.
    pub fn New2(parent: ?*anyopaque) QtC.QHPieModelMapper {
        return qtc.QHPieModelMapper_new2(@ptrCast(parent));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QHPieModelMapper_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qhpiemodelmapper.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Model(self: ?*anyopaque, ) QtC.QAbstractItemModel {
        return qtc.QHPieModelMapper_Model(@ptrCast(self));
    }

    pub fn SetModel(self: ?*anyopaque, model: ?*anyopaque) void {
        qtc.QHPieModelMapper_SetModel(@ptrCast(self), @ptrCast(model));
    }

    pub fn Series(self: ?*anyopaque, ) QtC.QPieSeries {
        return qtc.QHPieModelMapper_Series(@ptrCast(self));
    }

    pub fn SetSeries(self: ?*anyopaque, series: ?*anyopaque) void {
        qtc.QHPieModelMapper_SetSeries(@ptrCast(self), @ptrCast(series));
    }

    pub fn ValuesRow(self: ?*anyopaque, ) i32 {
        return qtc.QHPieModelMapper_ValuesRow(@ptrCast(self));
    }

    pub fn SetValuesRow(self: ?*anyopaque, valuesRow: i32) void {
        qtc.QHPieModelMapper_SetValuesRow(@ptrCast(self), valuesRow);
    }

    pub fn LabelsRow(self: ?*anyopaque, ) i32 {
        return qtc.QHPieModelMapper_LabelsRow(@ptrCast(self));
    }

    pub fn SetLabelsRow(self: ?*anyopaque, labelsRow: i32) void {
        qtc.QHPieModelMapper_SetLabelsRow(@ptrCast(self), labelsRow);
    }

    pub fn FirstColumn(self: ?*anyopaque, ) i32 {
        return qtc.QHPieModelMapper_FirstColumn(@ptrCast(self));
    }

    pub fn SetFirstColumn(self: ?*anyopaque, firstColumn: i32) void {
        qtc.QHPieModelMapper_SetFirstColumn(@ptrCast(self), firstColumn);
    }

    pub fn ColumnCount(self: ?*anyopaque, ) i32 {
        return qtc.QHPieModelMapper_ColumnCount(@ptrCast(self));
    }

    pub fn SetColumnCount(self: ?*anyopaque, columnCount: i32) void {
        qtc.QHPieModelMapper_SetColumnCount(@ptrCast(self), columnCount);
    }

    pub fn SeriesReplaced(self: ?*anyopaque, ) void {
        qtc.QHPieModelMapper_SeriesReplaced(@ptrCast(self));
    }

    pub fn OnSeriesReplaced(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QHPieModelMapper_Connect_SeriesReplaced(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn ModelReplaced(self: ?*anyopaque, ) void {
        qtc.QHPieModelMapper_ModelReplaced(@ptrCast(self));
    }

    pub fn OnModelReplaced(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QHPieModelMapper_Connect_ModelReplaced(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn ValuesRowChanged(self: ?*anyopaque, ) void {
        qtc.QHPieModelMapper_ValuesRowChanged(@ptrCast(self));
    }

    pub fn OnValuesRowChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QHPieModelMapper_Connect_ValuesRowChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn LabelsRowChanged(self: ?*anyopaque, ) void {
        qtc.QHPieModelMapper_LabelsRowChanged(@ptrCast(self));
    }

    pub fn OnLabelsRowChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QHPieModelMapper_Connect_LabelsRowChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn FirstColumnChanged(self: ?*anyopaque, ) void {
        qtc.QHPieModelMapper_FirstColumnChanged(@ptrCast(self));
    }

    pub fn OnFirstColumnChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QHPieModelMapper_Connect_FirstColumnChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn ColumnCountChanged(self: ?*anyopaque, ) void {
        qtc.QHPieModelMapper_ColumnCountChanged(@ptrCast(self));
    }

    pub fn OnColumnCountChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QHPieModelMapper_Connect_ColumnCountChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QHPieModelMapper_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qhpiemodelmapper.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QHPieModelMapper_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qhpiemodelmapper.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QHPieModelMapper_Delete(@ptrCast(self));
    }
};

File: binding/src/charts/libqhpiemodelmapper.cpp
#include <QAbstractItemModel>
#include <QHPieModelMapper>
#include <QObject>
#include <QPieModelMapper>
#include <QPieSeries>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <type_traits>
#include <qhpiemodelmapper.h>
#include "libqhpiemodelmapper.h"
#include "libqhpiemodelmapper.hxx"

QHPieModelMapper* QHPieModelMapper_new() {
	 return new QHPieModelMapper();
}

QHPieModelMapper* QHPieModelMapper_new2(QObject* parent) {
	 return new QHPieModelMapper(parent);
}

libqt_string QHPieModelMapper_Tr(const char* s) {
	QString _ret = QHPieModelMapper::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

QAbstractItemModel* QHPieModelMapper_Model(const QHPieModelMapper* self) {
	return self->model();
}

void QHPieModelMapper_SetModel(QHPieModelMapper* self, QAbstractItemModel* model) {
	self->setModel(model);
}

QPieSeries* QHPieModelMapper_Series(const QHPieModelMapper* self) {
	return self->series();
}

void QHPieModelMapper_SetSeries(QHPieModelMapper* self, QPieSeries* series) {
	self->setSeries(series);
}

int QHPieModelMapper_ValuesRow(const QHPieModelMapper* self) {
	return self->valuesRow();
}

void QHPieModelMapper_SetValuesRow(QHPieModelMapper* self, int valuesRow) {
	self->setValuesRow(valuesRow);
}

int QHPieModelMapper_LabelsRow(const QHPieModelMapper* self) {
	return self->labelsRow();
}

void QHPieModelMapper_SetLabelsRow(QHPieModelMapper* self, int labelsRow) {
	self->setLabelsRow(labelsRow);
}

int QHPieModelMapper_FirstColumn(const QHPieModelMapper* self) {
	return self->firstColumn();
}

void QHPieModelMapper_SetFirstColumn(QHPieModelMapper* self, int firstColumn) {
	self->setFirstColumn(firstColumn);
}

int QHPieModelMapper_ColumnCount(const QHPieModelMapper* self) {
	return self->columnCount();
}

void QHPieModelMapper_SetColumnCount(QHPieModelMapper* self, int columnCount) {
	self->setColumnCount(columnCount);
}

void QHPieModelMapper_SeriesReplaced(QHPieModelMapper* self) {
	self->seriesReplaced();
}

void QHPieModelMapper_Connect_SeriesReplaced(QHPieModelMapper* self, intptr_t slot) {
    void (*slotFunc)(QHPieModelMapper*) = reinterpret_cast<void (*)(QHPieModelMapper*)>(slot);
    QHPieModelMapper::connect(self, &QHPieModelMapper::seriesReplaced, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QHPieModelMapper_ModelReplaced(QHPieModelMapper* self) {
	self->modelReplaced();
}

void QHPieModelMapper_Connect_ModelReplaced(QHPieModelMapper* self, intptr_t slot) {
    void (*slotFunc)(QHPieModelMapper*) = reinterpret_cast<void (*)(QHPieModelMapper*)>(slot);
    QHPieModelMapper::connect(self, &QHPieModelMapper::modelReplaced, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QHPieModelMapper_ValuesRowChanged(QHPieModelMapper* self) {
	self->valuesRowChanged();
}

void QHPieModelMapper_Connect_ValuesRowChanged(QHPieModelMapper* self, intptr_t slot) {
    void (*slotFunc)(QHPieModelMapper*) = reinterpret_cast<void (*)(QHPieModelMapper*)>(slot);
    QHPieModelMapper::connect(self, &QHPieModelMapper::valuesRowChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QHPieModelMapper_LabelsRowChanged(QHPieModelMapper* self) {
	self->labelsRowChanged();
}

void QHPieModelMapper_Connect_LabelsRowChanged(QHPieModelMapper* self, intptr_t slot) {
    void (*slotFunc)(QHPieModelMapper*) = reinterpret_cast<void (*)(QHPieModelMapper*)>(slot);
    QHPieModelMapper::connect(self, &QHPieModelMapper::labelsRowChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QHPieModelMapper_FirstColumnChanged(QHPieModelMapper* self) {
	self->firstColumnChanged();
}

void QHPieModelMapper_Connect_FirstColumnChanged(QHPieModelMapper* self, intptr_t slot) {
    void (*slotFunc)(QHPieModelMapper*) = reinterpret_cast<void (*)(QHPieModelMapper*)>(slot);
    QHPieModelMapper::connect(self, &QHPieModelMapper::firstColumnChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QHPieModelMapper_ColumnCountChanged(QHPieModelMapper* self) {
	self->columnCountChanged();
}

void QHPieModelMapper_Connect_ColumnCountChanged(QHPieModelMapper* self, intptr_t slot) {
    void (*slotFunc)(QHPieModelMapper*) = reinterpret_cast<void (*)(QHPieModelMapper*)>(slot);
    QHPieModelMapper::connect(self, &QHPieModelMapper::columnCountChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

libqt_string QHPieModelMapper_Tr2(const char* s, const char* c) {
	QString _ret = QHPieModelMapper::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QHPieModelMapper_Tr3(const char* s, const char* c, int n) {
	QString _ret = QHPieModelMapper::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QHPieModelMapper_Delete(QHPieModelMapper* self) {
    delete self;
}

